include/mata/afa.hh

```diff
@@ -52,24 +61,88 @@ using Word = Mata::Nfa::Word;
 
 using StringDict = Mata::Nfa::StringDict;
 
+using StateSet = OrdVec<State>;
+
 using Alphabet = Mata::Nfa::Alphabet;
 
-/// A transition
+/*
+* A transition consists of a source state, a symbol and a vector of vector of states / vector of nodes
+* This corresponds to the formula in DNF
+*/
 struct Trans
 {
-	State src;
-  std::string formula;
+	State src; // source state
+    Symbol symb; // transition symbol
+    Nodes dst; // a vector of vectors of states
 
-	Trans() : src(), formula() { }
-	Trans(State src, const std::string& formula) : src(src), formula(formula) { }
+	Trans() : src(), symb(), dst() { }
+  	Trans(State src, Symbol symb, Node dst) : src(src), symb(symb), dst(Nodes(dst)) { }
+	Trans(State src, Symbol symb, Nodes dst) : src(src), symb(symb), dst(dst) { }
 
 	bool operator==(const Trans& rhs) const
 	{ // {{{
-		return src == rhs.src && formula == rhs.formula;
+		return src == rhs.src && symb == rhs.symb && dst == rhs.dst ;
 	} // operator== }}}
 	bool operator!=(const Trans& rhs) const { return !this->operator==(rhs); }
+
+    /*friend std::ostream& operator<<(std::ostream& os, const Trans trans)
+    {
+        return os << std::to_string(trans.src) + ":" + std::to_string(trans.symb) + "->" + std::to_string(trans.dst) + "\n";
+    }*/
+
 };
 
+using TransList = std::vector<Trans>;
+using TransRelation = std::vector<TransList>;
+
+/*
+* A tuple (result_nodes, sharing_list). If 'a' is an element of result_nodes, it means that there exists a symbol
+* 'b' such that sharing_list belongs to delta(a, b), where sharing_list is a node (vector of states). 
+*/
+typedef struct InvResults_{
+
+    Node result_nodes{}; 
+    Node sharing_list{};
+
+    InvResults_() : result_nodes(), sharing_list() { }
+    InvResults_(State state, Node sharing_list) : result_nodes(Node(state)), sharing_list(sharing_list) { }
+    InvResults_(Node result_nodes, Node sharing_list) : result_nodes(result_nodes), sharing_list(sharing_list) { }
+
+    bool operator==(InvResults_ rhs) const
+    { // {{{
+        return sharing_list == rhs.sharing_list && result_nodes == rhs.result_nodes;
+    } // operator== }}}
+    
+    bool operator!=(InvResults_ rhs) const
+    { // {{{
+        return !this->operator==(rhs);
+    } // operator!= }}}
+
+    bool operator<(InvResults_ rhs) const
+    { // {{{
+        return sharing_list < rhs.sharing_list || (sharing_list == rhs.sharing_list && result_nodes < rhs.result_nodes);
+    } // operator< }}}
+
+}InvResults;
+
+/*
+* A tuple (symb, vector_of_pointers). Each pointer points to the instance of InvResults. If &(result_nodes, sharing_list) belongs to the
+* vector_of_pointers, then for each 'a' which is part of result_nodes holds that 'sharing_list' is a member of delta(a, symbol).
+*/
+typedef struct InvTrans_{
+
+    using InvResultPtrs = std::vector<std::shared_ptr<InvResults>>;
```

Comment:
Thanks for your suggestion. I was thinking about these structures a lot and I realised that no pointers are necessary here.
Initially, I thought I needed to store each `InvResults` to the inverse transition relation multiple times to make it accessible from all the states of the corresponding `sharing_list`. Therefore I decided to use pointers to get rid of redundancy and avoid searching for all occurrences of the corresponding `InvResult` in case of changing its value. 
Then, I realised that I could choose one representative value of the `sharing_list` (minimal element of the set) and make the `InvResult` accessible in the transition relation only via this element so no pointers are needed anymore.